Userland/Libraries/LibGfx/ImageFormats/GIFLoader.cpp

```diff
@@ -183,7 +184,10 @@ static ErrorOr<void> decode_frame(GIFLoadingContext& context, size_t frame_index
         if (image->lzw_min_code_size > 8)
             return Error::from_string_literal("LZW minimum code size is greater than 8");
 
-        LZWDecoder decoder(image->lzw_encoded_bytes, image->lzw_min_code_size);
+        FixedMemoryStream stream { image->lzw_encoded_bytes, false };
```

Comment:
s:false:/*writing_enabled=*/false:

Reply:
If we want to be explicit about it, shouldn't we create a factory like `FixedMemoryStream::create_from_read_only_bytes`?

Reply:
I think factory is for fallible creation usually; here we'd normally use an enum arg instead of a bool if you want to do the extra brownie point fix 🙂 what I suggested is minimal more work and already kind of ok.

Reply:
Of course, how did I forget that!
I dislike how these inline comments look, and they don't age really well either, so if I can have an extra brownie point by not doing this, I'm all in :smile:.